sos/report/plugins/microshift.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2023 Red Hat, Inc. Pablo Acevedo <pacevedo@redhat.com>
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin, PluginOpt
+import re
+
+class Microshift(Plugin, RedHatPlugin):
+    """This is the plugin for MicroShift 4.X. Even though it shares some of
+    the OpenShift components, its IoT/Edge target makes the product nimble and
+    light, thus requiring different a approach when operating it.
+
+    When enabled, this plugin will collect cluster information (such as
+    systemd service logs, configuration, versions, etc.)and also inspect API
+    resources in certain namespaces. The namespaces to scan are kube.* and
+    openshift.*. Other namespaces may be collected by making use of the
+    ``only-namespaces`` or ``add-namespaces`` options.
+    """
+
+    short_desc = 'Microshift'
+    plugin_name = 'microshift'
+    plugin_timeout = 900
+    packages = (plugin_name,)
+    services = (plugin_name,)
+    profiles = (plugin_name,)
```

Comment:
Plugin profile is sort of name of a set of similar plugins like OpenStack, services, storage or similar. Why to introduce a new profile here?

FYI list of profiles and numbers of plugins assigned to them:

```
      1 "apache"
      1 "cifs"
      1 "mrg"
      1 "perl"
      3 "gpu"
      3 "java"
      3 "sap"
      4 "mail"
      4 "memory"
      4 "nfs"
      4 "openstack_undercloud"
      5 "openstack_compute"
      6 "debug"
      6 "performance"
      7 "kernel"
      7 "openshift"
      8 "packagemanager"
      9 "desktop"
     11 "boot"
     11 "webserver"
     13 "security"
     15 "cluster"
     15 "identity"
     21 "container"
     23 "openstack_controller"
     27 "sysmgmt"
     29 "openstack"
     31 "virt"
     34 "services"
     35 "network"
     59 "hardware"
     59 "storage"
    111 "system"
```

Reply:
That's the perfect use case for a preset. We can auto-load a specific configuration for sos report when `microshift` is present. No need for a specific config file or wrapper script or similar.

Users will just execute `sos report`, and behind the scenes the preset will set whatever options you want for it. Users can still override individual options from the preset via the CLI if needed.

Reply:
I am exploring the presets, will add it to the PR when its ready.